x-pack/plugins/security_solution/server/endpoint/services/actions/clients/lib/base_response_actions_client.ts

```diff
@@ -0,0 +1,366 @@
+        type: 'INPUT_ACTION',
+        input_type: actionRequest.agent_type ?? 'endpoint',
+        data: {
+          command: actionRequest.command,
+          comment: actionRequest.comment ?? undefined,
+          ...(actionRequest.alert_ids ? { alert_id: actionRequest.alert_ids } : {}),
+          ...(actionRequest.hosts ? { hosts: actionRequest.hosts } : {}),
+          parameters: getActionParameters(actionRequest) as EndpointActionDataParameterTypes,
+        },
+      },
+      user: {
+        id: this.options.username,
+      },
+      ...addRuleInfoToAction(actionRequest),
+    };
+
+    try {
+      const logsEndpointActionsResult = await this.options.esClient.index<LogsEndpointAction>(
+        {
+          index: ENDPOINT_ACTIONS_INDEX,
+          document: doc,
+          refresh: 'wait_for',
+        },
+        { meta: true }
+      );
+
+      if (logsEndpointActionsResult.statusCode !== 201) {
+        throw new ResponseActionsClientError(
+          `Failed to create (index) action request document. StatusCode: [${logsEndpointActionsResult.statusCode}] Result: ${logsEndpointActionsResult.body.result}`,
+          500,
+          logsEndpointActionsResult
+        );
+      }
+
+      return doc;
+    } catch (err) {
+      if (!(err instanceof ResponseActionsClientError)) {
+        throw new ResponseActionsClientError(
+          `Failed to create action request document: ${err.message}`,
+          500,
+          err
+        );
+      }
+
+      throw err;
+    }
+  }
+
+  /**
+   * Writes a Response Action response document to the Endpoint index
+   * @param options
+   * @protected
+   */
+  protected async writeActionResponseToEndpointIndex<TOutputContent extends object = object>({
+    actionId,
+    error,
+    agentId,
+    data,
+  }: ResponseActionsClientWriteActionResponseToEndpointIndexOptions<TOutputContent>): Promise<
+    LogsEndpointActionResponse<TOutputContent>
+  > {
+    const timestamp = new Date().toISOString();
+    const doc: LogsEndpointActionResponse<TOutputContent> = {
+      '@timestamp': timestamp,
+      agent: {
+        id: agentId,
+      },
+      EndpointActions: {
+        action_id: actionId,
+        started_at: timestamp,
+        completed_at: timestamp,
+        data,
+      },
+      error,
+    };
+
+    this.log.debug(`Writing response action response:\n${dump(doc)}`);
+
+    await this.options.esClient
+      .index<LogsEndpointActionResponse>({
+        index: ENDPOINT_ACTION_RESPONSES_INDEX,
+        document: doc,
+        refresh: 'wait_for',
+      })
+      .catch((err) => {
+        throw new ResponseActionsClientError(
+          `Failed to create action response document: ${err.message}`,
+          err.statusCode ?? 500,
+          err
+        );
+      });
+
+    return doc;
+  }
+
+  public async isolate(options: IsolationRouteRequestBody): Promise<ActionDetails> {
+    throw new ResponseActionsNotSupportedError('isolate');
```

Comment:
Another place where we hardcode action names

Reply:
I don't think we have an Enum for response actions